File: app/src/main/java/com/scatl/uestcbbs/module/credit/model/CreditModel.java
package com.scatl.uestcbbs.module.credit.model;

import com.scatl.uestcbbs.helper.rxhelper.Observer;
import com.scatl.uestcbbs.util.RetrofitCookieUtil;

import java.util.HashMap;
import java.util.Map;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class CreditModel {
    public void getMineCredit(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getMineCredit();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getCreditHistory(int page, int creditType, int inOrOut, Observer<String> observer) {
        Map<String, String> map = new HashMap<>();
        map.put("exttype", creditType + "");
        map.put("starttime", "");
        map.put("endtime", "");
        map.put("income", inOrOut + "");
        map.put("optype", "");
        map.put("search", "true");
        map.put("op", "log");
        map.put("ac", "credit");
        map.put("mod", "spacecp");

        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getCreditHistory(page, RetrofitCookieUtil.generateRequestBody(map));
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getCreditFormHash(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getCreditFormHash();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void creditTransfer(String formHash, String amount, String toUserName, String password, String message, Observer<String> observer) {
        Map<String, String> map = new HashMap<>();
        map.put("formhash", formHash);
        map.put("transfersubmit", "true");
        map.put("handlekey", "transfercredit");
        map.put("transferamount", amount);
        map.put("to", toUserName);
        map.put("password", password);
        map.put("transfermessage", message);
        map.put("transfersubmit_btn", "true");

        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .creditTransfer(RetrofitCookieUtil.generateRequestBody(map));
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getNewTask(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getNewTask();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getDoingTask(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getDoingTask();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getDoneTask(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getDoneTask();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getFailedTask(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getFailedTask();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getTaskDetail(int id, Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getTaskDetail(id);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void applyNewTask(int id, Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .applyNewTask(id);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void getTaskAward(int id, Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getTaskAward(id);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

    public void deleteDoingTask(int id, String formhash, Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .deleteDoingTask(id, formhash);
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);
    }

}

File: app/src/main/java/com/scatl/uestcbbs/module/credit/model/CreditModel.kt
package com.scatl.uestcbbs.module.credit.model

import com.scatl.uestcbbs.helper.rxhelper.Observer
import com.scatl.uestcbbs.util.RetrofitUtil
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Created by RetrofitUtil at 2023/6/20 14:32
 */
class CreditModel {

    fun getCreditHistory(page: Int, creditType: Int, inOrOut: Int): Observable<String> {
        val map = hashMapOf(
            "exttype"       to      creditType.toString(),
            "starttime"     to      "",
            "endtime"       to      "",
            "income"        to      inOrOut.toString(),
            "optype"        to      "",
            "search"        to      "true",
            "op"            to      "log",
            "ac"            to      "credit",
            "mod"           to      "spacecp",
        )
        
        return RetrofitUtil
            .getInstance()
            .apiService
            .getCreditHistory(page, RetrofitUtil.generateRequestBody(map))
    }

    fun getFormHash(): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .homeInfo
    }

    fun creditTransfer(formHash: String, amount: String, happyBoy: String, password: String, message: String): Observable<String> {
        val map = hashMapOf(
            "formhash"              to      formHash,
            "transfersubmit"        to      "true",
            "handlekey"             to      "transfercredit",
            "transferamount"        to      amount,
            "to"                    to      happyBoy,
            "password"              to      password,
            "transfermessage"       to      message,
            "transfersubmit_btn"    to      "true",
            "mod"                   to      "spacecp",
        )
        
        return RetrofitUtil
            .getInstance()
            .apiService
            .creditTransfer(RetrofitUtil.generateRequestBody(map))
    }

    fun getNewTask(): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .newTask
    }

    fun getDoingTask(): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .doingTask
    }

    fun getDoneTask(): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .doneTask
    }

    fun getFailedTask(): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .failedTask
    }

    fun getTaskDetail(id: Int): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .getTaskDetail(id)
    }

    fun applyNewTask(id: Int): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .applyNewTask(id)
    }

    fun getTaskAward(id: Int): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .getTaskAward(id)
    }

    fun deleteDoingTask(id: Int, formhash: String?): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .deleteDoingTask(id, formhash)
    }
}